R/validate.R

```diff
@@ -19,10 +20,25 @@
   assert_integerish(n)
   assert_matrix(triangle, all.missing = FALSE)
 
-  if (nrow(triangle) <= ncol(triangle)) {
+  # Check if the triangle has a valid structure
+  # Ensure each column has at least one non-NA value
+  if (any(colSums(!is.na(triangle)) == 0)) {
     cli_abort(
-      message =
-        "Number of observations(rows) must be greater than the number of columns." # nolint
+      message = c(
```

Comment:
This is a nice way of phrasing it that I never thought about

Reply:
We should add tests that this works (though I think this is on your list of new issues already)